substrate/pkg/controller/substrate/cluster/address.go

```diff
@@ -30,28 +33,52 @@ type Address struct {
 	EC2 *ec2.EC2
 }
 
+type addressOutput struct {
+	ipAddress    *string
+	allocationID *string
+}
+
 func (a *Address) Create(ctx context.Context, substrate *v1alpha1.Substrate) (reconcile.Result, error) {
-	addressesOutput, err := a.EC2.DescribeAddressesWithContext(ctx, &ec2.DescribeAddressesInput{Filters: discovery.Filters(substrate, discovery.Name(substrate))})
+	for _, req := range []struct {
+		name *string
+	}{
+		{name: discovery.Name(substrate, "apiserver")},
```

Comment:
we might want to document this behavior ? so customers ensure they don't duplicate these tags like `apiserver` and `natgw` outside of KIT Environment as you're discovering them below with these filters.